### ISSUE TITLE
Mipmap Example Shows Nearest Neighbor Sampling

### ISSUE THREAD

**Issue by cwfitzgerald** [gfx-rs/wgpu#1552]
**Description**

The mipmap reftests show non-linear sampling causing failure and a large amount of validation errors.

<details>
<summary>Errors</summary>

```
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0x4dbc9f000000001d[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0x3b4330000000001e[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0xd1a8a9000000001f[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0xee9f7d0000000020[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0xa2590f0000000021[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0x97b7a0000000022[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0x6401a90000000023[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroyImageView-imageView-01026 (0x63ac21f0)]
        Validation Error: [ VUID-vkDestroyImageView-imageView-01026 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x63ac21f0 | Cannot call vkDestroyImageView on VkImageView 0x2fa55c0000000024[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to imageView must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroyImageView-imageView-01026)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkDestroySampler-sampler-01082 (0xcca99934)]
        Validation Error: [ VUID-vkDestroySampler-sampler-01082 ] Object 0: handle = 0x1f569ee7d88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xcca99934 | Cannot call vkDestroySampler on VkSampler 0x799352000000001c[mip] that is currently in use by a command buffer. The Vulkan spec states: All submitted commands that refer to sampler must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkDestroySampler-sampler-01082)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DEVICE, hndl: 0x1f569ee7d88, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0x5e355b000000002c, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0x5e355b000000002c[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0x5e355b000000002c, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0x54508e0000000032, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0x54508e0000000032[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0x54508e0000000032, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0x4259cc000000002a, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0x4259cc000000002a[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0x4259cc000000002a, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0x43fa130000000030, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0x43fa130000000030[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0x43fa130000000030, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0x232af30000000036, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0x232af30000000036[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0x232af30000000036, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0x98e46c0000000027, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0x98e46c0000000027[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0x98e46c0000000027, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0x390827000000002e, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0x390827000000002e[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0x390827000000002e, name: ?)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance] VALIDATION [VUID-vkFreeDescriptorSets-pDescriptorSets-00309 (0xbfce1114)]
        Validation Error: [ VUID-vkFreeDescriptorSets-pDescriptorSets-00309 ] Object 0: handle = 0xf2d4f10000000034, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xbfce1114 | Cannot call vkFreeDescriptorSets() on VkDescriptorSet 0xf2d4f10000000034[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to any element of pDescriptorSets must have completed execution (https://vulkan.lunarg.com/doc/view/1.2.176.1/windows/1.2-extensions/vkspec.html#VUID-vkFreeDescriptorSets-pDescriptorSets-00309)
[2021-06-24T06:55:18Z ERROR wgpu_hal::vulkan::instance]         objects: (type: DESCRIPTOR_SET, hndl: 0xf2d4f10000000034, name: ?)
```

</details>

**Comment by cwfitzgerald** [comment]
VVL errors are gone but the image is still wrong.

**Comment by kvark** [comment]
Sorry about that! Fixed in 579de425e5b2aa861531db8b7f3d12e4fc3fb2ba
